### ISSUE TITLE
v1.28.0 shows v1.29.0-dev; v1.27.x tags all having 149 commits to current

### ISSUE THREAD

**Issue by swiffer** [teslamate-org/teslamate#3488]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

![grafik](https://github.com/teslamate-org/teslamate/assets/2990373/c63fed86-f414-4ee2-beca-602e49804c32)

### Expected Behavior

Should show v1.28.0

Why are all v1.27.x tags are showing 149 commits compared to latest? have branches diverged?
https://github.com/teslamate-org/teslamate/tags

### Relevant log output

```shell
-
```

### Type of installation

Docker

### Version

v1.28.0

**Comment by billerby** [comment]
Same here, also looks like the release action failed to publish the 1.28.0 image to docker.io (https://github.com/teslamate-org/teslamate/actions/runs/6988327850). Maybe related.

**Comment by rpearson-catdevnull** [comment]
See #3487. Think 1.28.0 has been marked as prerelease following that issue. Suspect once resolved should be fjne

**Comment by swiffer** [comment]
Tags v1.27.1, v1.27.2, v1.27.3 and v1.27.4 are based of commits from v1.27 branch. That branch has never been merged back into master - therefore clicking on tagged versions in CHANGELOG.md / compare tags is giving "wrong" results currently...

e.g.
https://github.com/teslamate-org/teslamate/compare/master...v1.27.3
https://github.com/teslamate-org/teslamate/compare/v1.27.3...master
https://github.com/teslamate-org/teslamate/compare/v1.27.4...v1.28.0

Don't we have to merge v1.19 into v1.20, v1.20 into v1.21, ... v.127 into master similar to how MariaDB is doing it?

https://github.com/MariaDB/server

**Comment by swiffer** [comment]
https://github.com/teslamate-org/teslamate/pull/3492

I would start like that - if you like I could to the other branches as well)

**Comment by JakobLichterfeld** [comment]
Good find, had oversexed it as well, will double-check tomorrow or on Monday.

**Comment by swiffer** [comment]
I don't get the branching model used up until now.

**What has happened:**

Whenever a new Minor Release is tagged a new Branch is created (e.g. v1.23). Changes happening afterwards (in master) have been copied into these release branches (not cherry-picked, different commits)

![grafik](https://github.com/teslamate-org/teslamate/assets/2990373/b88c589f-e5e9-428d-a254-a2bba1fa3562)

That means also those branches diverged from master but tags have been placed in these branches. Merging e.g. v1.20 into v1.21 now doesn't fix the the issue of not being able to compare tags correctly via git-diff

Tags could be deleted and recreated on the master branch instead but we shouldn't do that.
The moment we delete e.g. the v1.20 branch we are losing the tags, therefore we cannot do that either.

It might be best to keep everything the way it is and start working differently now (I would vote for having an ever-green master branch we are tagging new releases in and no longer create release branches). If we would like to do that we need to Merge before release and cherry-pick commits from master instead of somehow reapplying them.

**Comment by JakobLichterfeld** [comment]
> I don't see benefits in the branching model used up until now.

I could not agree more.

**Comment by swiffer** [comment]
@adriankumpf - could you rename master to main?

**Comment by JakobLichterfeld** [comment]
> @adriankumpf - could you rename master to main?

@cwanja do have the rights as well I assume.

**Comment by oivindoh** [comment]
Renaming master to main seems a bit.. pointless?

**Comment by cwanja** [comment]
@JakobLichterfeld - yes I have the ability to rename `master` (_shocked I somehow do, but alas here we are_). However, are they downstream net negative effects if I execute this? Or should it be performed in a controlled manner?

**Comment by cwanja** [comment]
Also, are we positive that `1.28.1` is safe to install now? Seen a handful of discussions and issues regarding it.

**Comment by swiffer** [comment]
Wait until Grafana Issues are sorted Out. I'll try to have a Look tomorrow morning...

**Comment by JakobLichterfeld** [comment]
> @JakobLichterfeld - yes I have the ability to rename `master` (_shocked I somehow do, but alas here we are_). However, are they downstream net negative effects if I execute this? Or should it be performed in a controlled manner?
>

Perfect, saw your repo rights in a list. It is safe to do so, but we need to adopt our CI pipelines manually, so we need to coordinate it.

**Comment by cwanja** [comment]
> > @JakobLichterfeld - yes I have the ability to rename `master` (_shocked I somehow do, but alas here we are_). However, are they downstream net negative effects if I execute this? Or should it be performed in a controlled manner?
> 
> Perfect, saw your repo rights in a list. It is safe to do so, but we need to adopt our CI pipelines manually, so we need to coordinate it.

Let me know how you want to coordinate @JakobLichterfeld in light of the comments above 😄

**Comment by JakobLichterfeld** [comment]
> However maybe it's best to Open a separate issue for that,

#3503
